### PR TITLE
Feature flags

### PR DESCRIPTION
**Pull request by adamweeks**

### CODE REVIEW

File: packages/node_modules/@ciscospark/redux-module-features/src/index.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+export default function reducer(state = initialState, action) {
+  switch (action.type) {
+  case ADD_FEATURE:
+    return state.mergeIn([`items`], action.payload.feature);
+  default:
+    return state;
+  }
+}
+
+export function addFeature(feature) {
```
Comment (bzang) [inline packages/node_modules/@ciscospark/redux-module-features/src/index.js]: Can we rename this to `storeFeature`? It took me a while to realize that this just saves the features we get from the server.

File: packages/node_modules/@ciscospark/redux-module-features/src/index.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+/**
+ * Gets the value of the requested feature toggle.
+ * @param {string} keyType <developer|user|entitlement>
+ * @param {string} key
+ * @param {object} spark
+ * @returns {Promise}
+ */
+export function getFeature(keyType, key, spark) {
```
Comment (bzang) [inline packages/node_modules/@ciscospark/redux-module-features/src/index.js]: Should we have a "getAllFeatures" method to grab all the features for a user? Or is the preferred way to fetch individual ones?

File: packages/node_modules/@ciscospark/redux-module-features/src/index.js (hunk 1 above)
Comment (adamweeks) [inline packages/node_modules/@ciscospark/redux-module-features/src/index.js]: Sure thing.

File: packages/node_modules/@ciscospark/redux-module-features/src/index.js (hunk 2 above)
Comment (adamweeks) [inline packages/node_modules/@ciscospark/redux-module-features/src/index.js]: @ianwremmel suggested to get specific features explicitly.

Comment (ianwremmel) [inline packages/node_modules/@ciscospark/redux-module-features/src/index.js]: yea, the full set is unbounded; i think i have several hundred at this point. the web client does copy them all from the sdk and wire up change listeners, but that's mostly there to provide a UI. if you only need to read them, getting them one at a time should be safer.

**Pull request merged**